core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -458,14 +457,14 @@ DeleteFileIndex build() {
     }
 
     private void add(
-        CharSequenceMap<PositionDeletes> deletesByPath,
+        Map<String, PositionDeletes> deletesByPath,
         PartitionMap<PositionDeletes> deletesByPartition,
         DeleteFile file) {
-      CharSequence path = ContentFileUtil.referencedDataFile(file);
+      String path = ContentFileUtil.referencedDataFileLocation(file);
 
       PositionDeletes deletes;
       if (path != null) {
-        deletes = deletesByPath.computeIfAbsent(path, PositionDeletes::new);
+        deletes = deletesByPath.computeIfAbsent(path, somePath -> new PositionDeletes());
```

Comment:
Minor: `somePath` -> `ignored` or `key`? We use this style in a few places, I think it highlights the key is not used.